### PR TITLE
Preliminar version

### PR DESCRIPTION
**Pull request by rolfsimoes**

### CODE REVIEW

**Comment by rolfsimoes** [comment]
.gitignore updated to *.so *.o files

**Comment by rolfsimoes** [comment]
Update documentation.

File: src/mrgs.cpp [hunk 1]
```diff
@@ -0,0 +1,540 @@
...
+//' ggplot(result) + 
+//'     geom_line(aes(resolution, 
+//'                   class_acc, 
+//'                   colour = as.factor(classes), 
+//'                   group = classes)) +
+//'     geom_line(aes(resolution, global_acc))
+//'
+//' @export
+// [[Rcpp::export]]
+Rcpp::DataFrame costanza(const Rcpp::NumericMatrix& x,
```
Comment (pedro-andrade-inpe) [inline src/mrgs.cpp]: `x` and `y` look like orthogonal variables, which is not the case. Instead of using `x` and `y` as arguments, use something like `data` and `reference`.

File: src/mrgs.cpp [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+                wx.move(i, j);
+                wy.move(i, j);
+                
+                for (int k = 0; k < vset_xy.size(); ++k) {
+                    count_x[vset_xy[k]] = 0;
+                    count_y[vset_xy[k]] = 0;
+                };
+                
+                for (int k = 0; k < wx.size(); ++k) {
+                    if (wx[k] == wx[k])
```
Comment (pedro-andrade-inpe) [inline src/mrgs.cpp]: It is better to implement a separate function (like `isnull(value)` or something like that) and use it here instead of verifying if a value is equals to itself.

File: src/mrgs.cpp (hunk 1 above)
Comment (rolfsimoes) [reply src/mrgs.cpp]: Updated in the last commit.

File: src/mrgs.cpp (hunk 2 above)
Comment (rolfsimoes) [reply src/mrgs.cpp]: Updated in the last commit.
